code/script.js

```diff
@@ -301,25 +462,57 @@ const filterCharacters = (keep) => {
   */
 
   // Invoke a function to redraw the board with the remaining people.
-}
+  generateBoard();
+};
+
+// Ylva - when clicking guess, the player first have to confirm that they want to make a guess.
+// If the user confirms the guess, the game will start and the checkMyGuess function will be invoked.
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
   // store the interaction from the player in a variable.
+const playersGuess = confirm(`Do you want to guess ${personToConfirm}?`);
   // remember the confirm() ?
+
   // If the player wants to guess, invoke the checkMyGuess function.
-}
+if (playersGuess) {
+checkMyGuess(personToConfirm);
+  }
+};
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
   // 1. Check if the personToCheck is the same as the secret person's name
   // 2. Set a Message to show in the win or lose section accordingly
   // 3. Show the win or lose section
   // 4. Hide the game board
-}
+  if (personToCheck === secret.name) {
+    // winningSound.play(); - might add sound here if time
+    winOrLoseText.innerText = `That's correct ${playerName}! The person is indeed ${secret.name}`;
+    secretImageAtCheck.innerHTML += `<img src=${secret.img} alt=${secret.name}>`;
+  } else {
+    // losingSound.play(); - might add a sound here if time
+    winOrLoseText.innerText = `You lost ${playerName}! It is not ${personToCheck}. The person is ${secret.name}`;
+    // winOrLoseSection.innerHTML += `
+    //     <img src=${secret.img} alt=${secret.name}>
+    // `;
+    secretImageAtCheck.innerHTML += `<img src=${secret.img} alt=${secret.name}>`;
+  }
+  winOrLoseSection.style.display = "flex";
```

Comment:
To make this look better, hide the other elements when showing the winOrLoseSection (set their style.display = "none")

Reply:
@edv-rd Great idea - thank you!